packages/server/src/api/controllers/row/alias.ts

```diff
@@ -0,0 +1,125 @@
+import { QueryJson, SearchFilters, Table, Row } from "@budibase/types"
+import { getDatasourceAndQuery } from "../../../sdk/app/rows/utils"
+import { cloneDeep } from "lodash"
+
+export default class AliasTables {
+  character: string
+  aliases: Record<string, string>
+  tableAliases: Record<string, string>
+  tableNames: string[]
+
+  constructor(tableNames: string[]) {
+    this.tableNames = tableNames
+    this.character = "a"
+    this.aliases = {}
+    this.tableAliases = {}
+  }
+
+  getAlias(tableName: string) {
+    if (this.aliases[tableName]) {
+      return this.aliases[tableName]
+    }
+    const char = this.character
+    this.aliases[tableName] = char
+    this.tableAliases[char] = tableName
+    this.character = String.fromCharCode(char.charCodeAt(0) + 1)
+    return char
+  }
+
+  aliasField(field: string) {
+    const tableNames = this.tableNames
+    if (field.includes(".")) {
+      const [tableName, column] = field.split(".")
+      if (tableNames.includes(tableName)) {
+        return `${this.getAlias(tableName)}.${column}`
+      }
+    }
+    return field
+  }
+
+  reverse<T extends Row | Row[]>(rows: T): T {
+    const process = (row: Row) => {
+      const final: Row = {}
+      for (let [key, value] of Object.entries(row)) {
+        if (!key.includes(".")) {
+          final[key] = value
+        } else {
+          const [alias, column] = key.split(".")
+          const tableName = this.tableAliases[alias] || alias
+          final[`${tableName}.${column}`] = value
+        }
+      }
+      return final
+    }
+    if (Array.isArray(rows)) {
+      return rows.map(row => process(row)) as T
+    } else {
+      return process(rows) as T
+    }
+  }
+
+  aliasMap(tableNames: (string | undefined)[]) {
+    const map: Record<string, string> = {}
+    for (let tableName of tableNames) {
+      if (tableName) {
+        map[tableName] = this.getAlias(tableName)
+      }
+    }
+    return map
+  }
+
+  async queryWithAliasing(json: QueryJson) {
+    json = cloneDeep(json)
```

Comment:
We do a deep clone here and then proceed to do quite a lot more internal cloning later with spread operators. I don't imagine it's going to be a massive performance hit, but does feel quite wasteful. No real action here, other than if it's easy to do so, it'd be nice for this to be the only copy made of `json`.